### PR TITLE
Fix incorrect regular expressions in expect test

### PR DESCRIPTION
**Pull request by tsachiherman**
## Summary

Explain the goal of this change and what problem it is solving. Format this cleanly so that it may be used for a commit message, as your changes will be squash-merged.

## Test Plan

How did you test these changes? Please provide the exact scenarios you tested in as much detail as possible including commands, output and rationale.

### CODE REVIEW

File: test/e2e-go/cli/goal/expect/goalAppAccountAddressTest.exp [hunk 1]
```diff
@@ -148,7 +148,7 @@ proc goalAppAccountAddress { TEST_ALGO_DIR TEST_DATA_DIR} {
 	--app-account $ACCOUNT_4_ADDRESS
     expect {
 	timeout { puts timeout; ::AlgorandGoal::Abort  "\n Failed to see expected output" }
-	"*Couldn't broadcast tx with algod: HTTP 400 Bad Request: TransactionPool.Remember: transaction*invalid Accounts index 4*" \
+	"^Couldn't broadcast tx with algod: HTTP 400 Bad Request: TransactionPool\.Remember: transaction.*invalid Accounts index 4.*" \
```
Comment (algonautshant) [inline test/e2e-go/cli/goal/expect/goalAppAccountAddressTest.exp:151]: `\.` is not necessary. `.` is sufficient. `?` represent `any one character`. `.` will be exact match to `.`.

File: test/e2e-go/cli/goal/expect/goalAppAccountAddressTest.exp [hunk 2]
```diff
@@ -112,19 +112,19 @@ proc goalAppAccountAddress { TEST_ALGO_DIR TEST_DATA_DIR} {
     ::AlgorandGoal::CheckProcessReturnedCode 1
 
     puts "Checking the results"
-    set EXPECTED_OUTPUT "Account0*$PRIMARY_ACCOUNT_ADDRESS"
-    set EXPECTED_OUTPUT "$EXPECTED_OUTPUT*Account1*$ACCOUNT_1_ADDRESS"
-    set EXPECTED_OUTPUT "$EXPECTED_OUTPUT*Account2*$ACCOUNT_2_ADDRESS"
-    set EXPECTED_OUTPUT "$EXPECTED_OUTPUT*Account3*$ACCOUNT_3_ADDRESS"
-    set EXPECTED_OUTPUT "$EXPECTED_OUTPUT*Account4*$ACCOUNT_4_ADDRESS"
-    set EXPECTED_OUTPUT "$EXPECTED_OUTPUT*Num Accounts*ui*4"
+    set EXPECTED_OUTPUT "Account0.*$PRIMARY_ACCOUNT_ADDRESS"
```
Comment (algonautshant) [inline test/e2e-go/cli/goal/expect/goalAppAccountAddressTest.exp:115]: The placeholder for `any once character` is `?`. `.` will match exactly `.`
Substituting`.*` with `?*` will fix the failure, and will evaluate one or more characters as expected.

**Review by algonautshant — APPROVED** [review]
Looks good. 
I provided comments on how to fix the failure.

**Comment by tsachiherman** [comment]
I'm closing this one, as it wasn't a regular expression at all - it was a glob pattern.
As for a glob pattern, its functionally correct, so I wouldn't want to touch that.

**Pull request closed without merging**